crates/python/src/api.rs

```diff
@@ -88,6 +91,7 @@ create_exception!(qcs, TranslationError, PyRuntimeError);
 create_exception!(qcs, CompilationError, PyRuntimeError);
 create_exception!(qcs, RewriteArithmeticError, PyRuntimeError);
 create_exception!(qcs, DeviceIsaError, PyValueError);
+create_exception!(qcs, QcsApiError, PyRuntimeError);
```

Comment:
Along with using a non-boxed Rust error, the `wrap_error`/`py_wrap_error` method should be used here instead.